contracts/modules/TransferManager/MaximumVolumeTransferManager.sol

```diff
@@ -0,0 +1,296 @@
+            _maximumVolumes.length == _endTimes.length &&
+            _maximumVolumes.length == _rollingPeriodIntervals.length,
+            "Input array length mismatch"
+        );
+
+        for(uint256 i = 0; i < _maximumVolumes.length; i++){
+            addMaxVolumeRestricion(
+                _maximumVolumes[i],
+                _startTimes[i],
+                _endTimes[i],
+                _rollingPeriodIntervals[i]
+            );
+        }
+    }
+
+    /**
+    * @notice allows admin to modify a maximum volume transfer restriction
+    * @param _index representing the array index of the maximum volume transfer restriction
+    * @param _maximumVolume maximum token volume that can be transfered
+    * @param _startTime timestamp representing when this maximum volume restriction starts
+    * @param _endTime timestamp representing when this maximum volume restriction ends
+    * @param _rollingPeriodInterval interval period representing one of five RollingIntervals
+    */
+    function modifyMaximumTransferRestriction(
+        uint256 _index, 
+        uint256 _maximumVolume, 
+        uint256 _startTime, 
+        uint256 _endTime, 
+        uint256 _rollingPeriodInterval) 
+    public withPerm(ADMIN) {
+        require(_index < maximumVolumeRestrictions.length, "invalid index");
+        require(now < maximumVolumeRestrictions[_index].endTime, "restriction is no longer active");
+        _checkRestrictionParams(_startTime, _endTime, _rollingPeriodInterval);
+
+        if(_startTime == 0) {
+            _startTime = now;
+            require(_endTime > _startTime, "end time must be greater than start time");
+        }
+
+        MaximumVolumeRestriction memory restriction = MaximumVolumeRestriction(
+            _maximumVolume, 
+            _startTime, 
+            _endTime, 
+            RollingIntervals(_rollingPeriodInterval)
+        );
+        maximumVolumeRestrictions[_index] = restriction;
+
+        emit ModifyMaximumVolumeRestriction(
+            _index,
+            now, 
+            _maximumVolume,
+            _startTime,
+            _endTime,
+            _rollingPeriodInterval
+        );
+    }
+
+    /**
+    * @notice allows admin to delete a maximum volume transfer restriction
+    * @param _index representing the array index of the maximum volume transfer restriction
+    */
+    function removeMaximumTransferRestriction(uint256 _index) public withPerm(ADMIN){
+        require (maximumVolumeRestrictions.length != 0, "no existing restrictions");
+        require (maximumVolumeRestrictions.length > _index, "invalid index");
+
+        if(_index < (maximumVolumeRestrictions.length - 1)){
+            MaximumVolumeRestriction memory lastRestriction = maximumVolumeRestrictions[maximumVolumeRestrictions.length - 1];
+            maximumVolumeRestrictions[_index] = lastRestriction;
+        }
+
+        maximumVolumeRestrictions.length = (maximumVolumeRestrictions.length).sub(1);
```

Comment:
SafeMath is not required here as it will revert before reaching this statement if `maximumVolumeRestrictions.length == 0`
```suggestion
        maximumVolumeRestrictions.length = maximumVolumeRestrictions.length - 1;
```